scripts/16_create-codebook.R

```diff
@@ -109,225 +141,49 @@ codebook <-
-#
-#   "n_reg_pub_doi_or_pmid",
-#   "",
-#
-#   "results_followup",
-#   "",
-#
-#   "has_followup_2y",
-#   "",
-#
-#   "has_followup_5y",
-#   "",
-#
-#   "is_summary_results_1y",
-#   "",
-#
-#   "is_summary_results_2y",
-#   "",
-#
-#   "is_summary_results_5y",
-#   "",
-#
-#   "is_publication_2y",
-#   "",
-#
-#   "is_publication_5y",
-#   "",
-#
-#   "color_green_only",
-#   "",
-#
-#   "color",
-#   "",
-#
-#   "issn",
-#   "",
-#
-#   "journal_unpaywall",
-#   "",
-#
-#   "publisher",
-#   "",
-#
-#   "publication_date_unpaywall",
-#   "",
-#
-#   "syp_response",
-#   "",
-#
-#   "can_archive",
-#   "",
-#
-#   "archiving_locations",
-#   "",
-#
-#   "inst_repository",
-#   "",
-#
-#   "versions",
-#   "",
-#
-#   "submitted_version",
-#   "",
-#
-#   "accepted_version",
-#   "",
-#
-#   "published_version",
-#   "",
-#
-#   "licenses_required",
-#   "",
-#
-#   "permission_issuer",
-#   "",
-#
-#   "embargo",
-#   "",
-#
-#   "date_embargo_elapsed",
-#   "",
-#
-#   "is_embargo_elapsed",
-#   "",
-#
-#   "permission_accepted",
-#   "",
-#
-#   "permission_published",
-#   "",
-#
-#   "is_oa",
-#   "",
-#
-#   "is_archivable",
-#   "",
-#
-#   "is_closed_archivable",
-#   ""
-# ) |>
-#
-#   # Add intovalue descriptions
-#   bind_rows(iv_codebook)
-#
-# # Check that all described variables are in the dataset and vice-versa
-# if (!all(colnames(trials) %in% description$name)| !all(description$name %in% colnames(trials))){
-#   rlang::warn("There are missing or additional variables for codebook!")
-# }
+  # Add original iv descriptions
+  left_join(iv_codebook, by = "name")
 
+write_csv(codebook_structure, codebook_structure_path)
+
+
+# 3. Manually edit codebook -----------------------------------------------
+# Manually upload `codebook-structure.csv` to google spreadsheet to collaboratively add and edit `definition`, as well as add each variable's `source` and whether it `is_derived`.
+
+
+# 4. Build codebook with descriptions -------------------------------------
+
+# Use google identity (i.e., gmail) to access for google sheets
+# Get google identity if locally stored as "google", if available
+# Else ask user and store
+google_id <-
+  ifelse(
+    nrow(keyring::key_list("google")) == 1,
+    keyring::key_get("google"),
+    keyring::key_set("google")
+  )
+
+message("Accessing googlesheets via: ", google_id)
+
+# If new google identity, prompt user in web browser to authenticate
+googlesheets4::gs4_auth(google_id)
+
+codebook_descriptions <-
+  googlesheets4::read_sheet("https://docs.google.com/spreadsheets/d/1h9MyoeaP4owH4-5ySLob0RMEuud5omKOLMeIY8C4-UE/edit#gid=807502810", sheet = "codebook")
+
+# Combine codebook structure and descriptions
 codebook <-
-  codebook |>
-  # left_join(description, by = "name") |>
-  left_join(iv_codebook, by = "name") |>
-  relocate(description, .after = "type")
+  left_join(
+    select(codebook_structure, -description),
+    select(codebook_descriptions, name, description, source, is_derived)
+  )
```

Comment:
i did it this way as defensive programming — my attitude is generally that I will make a typo so minimize data that has gone through me (or other variables, such as other humans, google sheets, etc.). Since `codebook_descriptions` could in theory have had it's variables messed with, I had preferenced the data that stayed put.

I did however try just now:
```r
codebook_new <-
  left_join(
    select(codebook_descriptions, name, type, levels, description, source, is_derived),
    select(codebook_structure, -description),
    by = c("name", "type", "levels")
  )
waldo::compare(codebook, codebook_new)
```

Seems to be pretty clean, but a few differences which I'm concerned are character encodings.

I'd be inclined to leave as is but let me know if you prefer to make the switch.

Reply:
@maia-sh you're right, the way you did it makes sense. I'm a bit puzzled by the diffs when running the code above, as it doesn't just seem to be character encoding. Some of the levels just seem to have a different order, e.g., center size: `No lead center; large; small` and `large; No lead center; small`. But your approach makes sense, let's keep it as is.